test/release/examples/primers/interopWithC.chpl

```diff
@@ -0,0 +1,332 @@
+   generated ``lib`` directory.
+   Since ``--static`` is not supported on MacOS you will have to dynamically link
+   the generated library. Please refer to :ref:`readme-libraries` for instructions
+   on how to do that.
+*/
+
+/*
+   .. _primers-C-interop-using-library:
+
+   Using the Chapel Library
+   ++++++++++++++++++++++++
+*/
+
+/*
+   To access the symbols in the generated Chapel library from a C program,
+   ``#include`` the generated header file:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 1-2
+*/
+
+/*   
+   Before calling any of the functions defined by the library, the Chapel
+   runtime and internal libraries must be initialized.  This is done by calling
+   ``chpl_library_init``, which is defined in
+   ``$CHPL_HOME/runtime/include/chpl-init.h`` and accessible with the above
+   ``#include``.
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 5-7
+*/
+
+/*   
+   Then, if the exported functions rely upon any global variables, the module
+   initialization function declared in the header file must be called.  If there
+   is uncertainty about whether doing so is necessary, it is recommended to call
+   this function.  In this example, this call occurs like this:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 8-9
+*/
+
+//
+// After that, the exported functions may be called:
+//
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 11-13
+*/
+
+/*
+   When ending the C program, the user must explicitly shut down the Chapel
+   runtime and module code.  This is done by calling ``chpl_library_finalize``:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 15-19
+*/
+
+/*
+   Once shut down, the Chapel runtime cannot be restarted in the current C
+   program.
+*/
+
+/*
+   .. _primers-C-interop-compiling-2:
+
+   Compiling C Code with the Chapel Library
+   ++++++++++++++++++++++++++++++++++++++++
+*/
+
+/*
+   Compiling C code with the generated Chapel library file is generally complex,
+   but can be made simpler through the use of the ``--library-makefile`` flag
+   as described in :ref:`readme-libraries-linking`.
+   
+   An example of compiling a C program with a generated Chapel library using the
+   generated Makefile can be found in the `Makefile
+   <https://github.com/chapel-lang/chapel/blob/master/test/release/examples/primers/Makefile>`_
+   for the primers directory, to build this source file.
+*/ 
+
+/*
+   .. _primers-C-interop-using-C:
+
+   Using C Code in Chapel
+   ----------------------
+*/
+
+/*
+   Chapel has support for C code but we need to tell the compiler
+   about the C symbols using the ``extern`` keyword
```

Comment:
Let's end this sentence with a period :) It'd also be nice to explicitly mention that we're using the keyword with a function.

Reply:
I used the word symbol because `extern` keyword is not limited to functions. It can also apply to variables and structs. 
Should I use something other than that?

Reply:
You do cover that later in the doc, but it's okay to leave it in